lib/app/services/database/conversation_db_service.c.dart

```
@@ -0,0 +1,345 @@
+          ConversationMessagesTableData(
+            isDeleted: isDeleted,
+            conversationId: conversationId,
+            eventMessageId: conversationMessage.id,
+            createdAt: conversationMessage.createdAt,
+            pubKeys: conversationMessage.allPubkeysMask,
+            subject: conversationMessage.data.relatedSubject?.value,
+            status: DeliveryStatus.none,
+          ),
+          mode: InsertMode.insertOrReplace,
+        );
+  }
+
+  Future<int> _insertConversationReactionsTableData(EventMessage eventMessage) {
+    final reactionEntity = PrivateMessageReactionEntity.fromEventMessage(eventMessage);
+
+    return _db.into(_db.conversationReactionsTable).insert(
+          ConversationReactionsTableData(
+            reactionEventId: reactionEntity.id,
+            createdAt: reactionEntity.createdAt,
+            content: reactionEntity.data.content,
+            messageId: reactionEntity.data.eventId,
+          ),
+          mode: InsertMode.insertOrReplace,
+        );
+  }
+
+  Future<void> insertEventMessage(EventMessage eventMessage) async {
+    await _db.into(_db.eventMessagesTable).insert(
+          EventMessageTableData.fromEventMessage(eventMessage),
+          mode: InsertMode.insertOrReplace,
+        );
+
+    if (eventMessage.kind == PrivateDirectMessageEntity.kind) {
+      await _updateConversationMessagesTable(eventMessage);
+    } else if (eventMessage.kind == PrivateMessageReactionEntity.kind) {
+      switch (eventMessage.content) {
+        case 'received':
+          await _updateConversationMessageAsReceived(eventMessage);
+        case 'read':
+          await _updateConversationMessagesAsRead(eventMessage);
+        default:
+          await _insertConversationReactionsTableData(eventMessage);
+      }
+    }
+  }
+
+  Future<void> _updateConversationMessagesTable(
+    EventMessage eventMessage,
+  ) async {
+    if (eventMessage.kind == PrivateDirectMessageEntity.kind) {
```

Use `return fast` or `fail fast` to reduce indentation. Like `if (eventMessage.kind != PrivateDirectMessageEntity.kind) return;`

---

Done